crates/bevy_ui/src/stack.rs

```diff
@@ -55,6 +56,14 @@ pub fn ui_stack_system(
     ui_stack.uinodes.clear();
     ui_stack.uinodes.reserve(total_entry_count);
     fill_stack_recursively(&mut ui_stack.uinodes, &mut global_context);
+
+    for (i, entity) in ui_stack.uinodes.iter().enumerate() {
+        update_query
+            .get_mut(*entity)
+            .unwrap()
+            .bypass_change_detection()
+            .stack_index = i as u32;
+    }
```

Comment:
Instead of having an extra update query you could have `fill_stack_recursively` set the index, e.g. by taking a `&mut u32` that it uses and increments.

Reply:
Ah it doesn't have the node. I don't know enough about this system to know if that's easy to get on one of the other queries, otherwise this is probably the simplest solution.

Reply:
Yep there might be more efficient ways of updating the `Node` components. I didn't want to make anything more than minor changes with this PR though. The improvement in extraction performance is good enough on its own.